src/sdk/k4a.c

```diff
@@ -526,171 +526,252 @@ static k4a_result_t validate_configuration(k4a_context_t *device, const k4a_devi
 {
     RETURN_VALUE_IF_ARG(K4A_RESULT_FAILED, config == NULL);
     RETURN_VALUE_IF_ARG(K4A_RESULT_FAILED, device == NULL);
-    k4a_result_t result;
+    k4a_result_t result = K4A_RESULT_SUCCEEDED;
     bool depth_enabled = false;
     bool color_enabled = false;
 
-    result = K4A_RESULT_FROM_BOOL(
-        config->color_format == K4A_IMAGE_FORMAT_COLOR_MJPG || config->color_format == K4A_IMAGE_FORMAT_COLOR_YUY2 ||
-        config->color_format == K4A_IMAGE_FORMAT_COLOR_NV12 || config->color_format == K4A_IMAGE_FORMAT_COLOR_BGRA32);
-    if (K4A_SUCCEEDED(result))
+    if (config->color_format != K4A_IMAGE_FORMAT_COLOR_MJPG && config->color_format != K4A_IMAGE_FORMAT_COLOR_YUY2 &&
+        config->color_format != K4A_IMAGE_FORMAT_COLOR_NV12 && config->color_format != K4A_IMAGE_FORMAT_COLOR_BGRA32)
     {
-        result = K4A_RESULT_FROM_BOOL(config->color_resolution >= K4A_COLOR_RESOLUTION_OFF &&
-                                      config->color_resolution <= K4A_COLOR_RESOLUTION_3072P);
+        result = K4A_RESULT_FAILED;
+        LOG_ERROR("The configured color format %d is not supported.", config->color_format);
     }
+
     if (K4A_SUCCEEDED(result))
     {
-        result = K4A_RESULT_FROM_BOOL(config->depth_mode >= K4A_DEPTH_MODE_OFF &&
-                                      config->depth_mode <= K4A_DEPTH_MODE_PASSIVE_IR);
+        if (config->color_resolution < K4A_COLOR_RESOLUTION_OFF ||
+            config->color_resolution > K4A_COLOR_RESOLUTION_3072P)
+        {
+            result = K4A_RESULT_FAILED;
+            LOG_ERROR("The configured color resolution %d is not supported.", config->color_resolution);
+        }
     }
+
     if (K4A_SUCCEEDED(result))
     {
-        result = K4A_RESULT_FROM_BOOL(config->camera_fps == K4A_FRAMES_PER_SECOND_5 ||
-                                      config->camera_fps == K4A_FRAMES_PER_SECOND_15 ||
-                                      config->camera_fps == K4A_FRAMES_PER_SECOND_30);
+        if (config->depth_mode < K4A_DEPTH_MODE_OFF || config->depth_mode > K4A_DEPTH_MODE_PASSIVE_IR)
+        {
+            result = K4A_RESULT_FAILED;
+            LOG_ERROR("The configured color depth mode %d is not supported.", config->depth_mode);
+        }
     }
+
     if (K4A_SUCCEEDED(result))
     {
-        result = K4A_RESULT_FROM_BOOL(config->wired_sync_mode >= K4A_WIRED_SYNC_MODE_STANDALONE &&
-                                      config->wired_sync_mode <= K4A_WIRED_SYNC_MODE_SUBORDINATE);
+        if (config->camera_fps != K4A_FRAMES_PER_SECOND_5 && config->camera_fps != K4A_FRAMES_PER_SECOND_15 &&
+            config->camera_fps != K4A_FRAMES_PER_SECOND_30)
+        {
+            result = K4A_RESULT_FAILED;
+            LOG_ERROR("The configured camera fps %d is not supported.", config->camera_fps);
```

Comment:
The previous comment is especially important here, where if they passed in a value 30 for example, they would get a message:
"The configured camera fps 30 is not supported"

Which would be confusing since K4A_FRAMES_PER_SECOND_30 is supported.